src/api.py

```diff
@@ -1,26 +1,100 @@
-from flask import Flask, request
+import os
+
+from celery.result import AsyncResult
+from fastapi import FastAPI
+from fastapi.params import Query
+from pydantic import BaseModel
+from starlette.responses import HTMLResponse
+
 from src import service
+from src.generators import html_generator
+
+app = FastAPI()
+format_html = False
+
 
-app = Flask(__name__)
+class TaskOut(BaseModel):
+    id: str
+    status: str
+    result: str | None = None
+    status_callback_url: str | None = None
 
 
 # Sample url: https://tmt.org/?test-url=https://github.com/teemtee/tmt&test-name=/tests/core/smoke
 # or for plans: https://tmt.org/?plan-url=https://github.com/teemtee/tmt&plan-name=/plans/features/basic
-@app.route("/", methods=["GET"])
-def find_test():
-    test_url = request.args.get("test-url", default=None)
-    test_name = request.args.get("test-name", default=None)
-    test_ref = request.args.get("test-ref", default="main")
+@app.get("/")
+def find_test(
+        test_url: str = Query(None, alias="test-url"),
+        test_name: str = Query(None, alias="test-name"),
+        test_ref: str = Query("default", alias="test-ref"),
+        test_path: str = Query(None, alias="test-path"),
+        plan_url: str = Query(None, alias="plan-url"),
+        plan_name: str = Query(None, alias="plan-name"),
+        plan_ref: str = Query("default", alias="plan-ref"),
+        plan_path: str = Query(None, alias="plan-path"),
+        out_format: str = Query("json", alias="format")
```

Comment:
to specify a list accepted values:

```suggestion
        out_format: Annotated[Literal["html", "json"], Query(alias="format")] = "json",
```

Reply:
Here adding `from typing import Literal` is required too.